cmd/profile/add/add.go

```diff
@@ -175,18 +176,24 @@ func LoadProfilesFromFile(filePath string) ([]models.DeviceProfile, error) {
 	}
 
 	var profiles []models.DeviceProfile
-
-	//check if the file contains just one DeviceProfile
 	var p models.DeviceProfile
-	err = json.Unmarshal(file, &p)
+	//First check if the file contains Yaml
```

Comment:
Would it not make sense to use the file extension to determine whether we need to unmarshall as json or yaml. (check of yml and yaml as valid extensions).
Already we need to check for errors in case there are multiple profiles defined in the file versus one.

Reply:
Nowhere in the CLI I have check about the type of the file extension.The clients are free to name their files however they want. But because Device Profile is the only exception that supports not just Json but Yaml as well I  suggest to brute force it - try the one , if fails try the other ..or if you think it is bad idea :) to add some additional flag specifying the type of the file instead of forcing the client to name their file with proper extensions.  Example: 
./edgex-cli profile add -f <filename>  - > by default the command will try to parse the file data using json (to preserve consistency with the others commands)
./edgex-cli profile add - f <filename> --yaml -> try to parse the content of the provided file in Yaml